easybuild/tools/modules.py

```diff
@@ -674,20 +674,33 @@ def read_module_file(self, mod_name):
     def modpath_extensions_for(self, mod_names):
         """
         Determine dictionary with $MODULEPATH extensions for specified modules.
-        Modules with an empty list of $MODULEPATH extensions are included.
+        All potential $MODULEPATH extensions are included, even the ones guarded by a condition (which is not checked).
+        Only direct $MODULEPATH extensions are found, no recursion if performed for modules that load other modules.
+        Modules with an empty list of $MODULEPATH extensions are included in the result.
+
+        @param mod_names: list of module names for which to determine the list of $MODULEPATH extensions
+        @return: dictionary with module names as keys and lists of $MODULEPATH extensions as values
         """
         self.log.debug("Determining $MODULEPATH extensions for modules %s" % mod_names)
 
         # copy environment so we can restore it
         env = os.environ.copy()
 
+        # regex for $MODULEPATH extensions;
+        # via 'module use ...' or 'prepend-path MODULEPATH' in Tcl modules,
+        # or 'prepend_path("MODULEPATH", "...") in Lua modules
+        modpath_ext_regex = r'|'.join([
+            r'^\s*module\s+use\s+"?([^"\s]+)"?',  # 'module use' in Tcl module files
+            r'^\s*prepend-path\s+MODULEPATH\s+"?([^"\s]+)"?',  # prepend to $MODULEPATH in Tcl modules
+            r'^\s*prepend_path\(\"MODULEPATH\",\s*\"(\S+)\"',  # prepend to $MODULEPATH in Lua modules
+        ])
+        modpath_ext_regex = re.compile(modpath_ext_regex, re.M)
```

Comment:
Maybe, but what extra would that give us?

If we detect a Tcl module (which is easy, via the required `%#Module` header), then we still need to try two regex's: `module use` and `prepend-path MODULEPATH`...

It's not like this is going to result in a noticeable speedup: module files are quite small, and regex search in Python is quite fast. It may even be faster to do what we do now than figure out first which type of module we're dealing with and then searching with a regex...